test/e2e/autoscaling/horizontal_pod_autoscaling_behavior.go

```diff
@@ -378,4 +379,74 @@ var _ = SIGDescribe("[Feature:HPA] [Serial] [Slow] Horizontal pod autoscaling (n
 			framework.ExpectEqual(timeWaitedFor2 < deadline, true, "waited %s, wanted less than %s", timeWaitedFor2, deadline)
 		})
 	})
+
+	ginkgo.Describe("with both scale up and down controls configured", func() {
+		ginkgo.It("should keep recommendation within the range over two stabilization windows", func() {
```

Comment:
I don't quite understand what does this scenario verify. We already have a test for scale up stabilization window, right? What's the difference here?

Reply:
Have a look at this issue for context: https://github.com/kubernetes/kubernetes/issues/96671.

This first test is reproducing error scenario 1 from that issue:
```
stabilizationWindowSeconds set up to 5 minutes for both scale up and scale down
min replica count 2
Application consumes big amount of resources during warm-up
0 minute: current replica count: 2
1 minute: desired replica count: 3 - no scale up stabilization window worked
2 minute: desired replica count: 1 (!) - scale up to 3 (max number from previous recommendations during stabilization window period)
```

This was a HPA problem when both up and down scale stabilisation is configured. There is a nice explanation with images from @josephburnett here: https://github.com/kubernetes/kubernetes/issues/96671#issuecomment-746283934

Reply:
And the requirement for that test was mentioned by Joe here: https://github.com/kubernetes/kubernetes/issues/110172 under this point:
```scale and and down controls (see https://github.com/kubernetes/kubernetes/pull/97348))```

Reply:
Oh, thanks for explaining.

But you're not really verifying that the recommendation is kept within range, right? Just how long it took to scale to 6 pods.

Reply:
After consideration - I've modifed this to be more precise:
1. We increase CPU usage and check if replicas stay in the desired window.
2. We wait for the scale up to finish after the window is over.
3. We decrease CPU usage and check if replicas stay in desired window.
4. We wait for the scale down to finish after window is over.

With such approach we verify both that the range is kept and that target replicas are met.